app/src/main/java/com/elmeradrianv/shesafe/fragments/ProfileUserFragment.java

```diff
@@ -0,0 +1,275 @@
+import com.elmeradrianv.shesafe.LoginActivity;
+import com.elmeradrianv.shesafe.R;
+import com.elmeradrianv.shesafe.adapters.EmergencyContactsCardAdapter;
+import com.elmeradrianv.shesafe.auxiliar.MyPair;
+import com.elmeradrianv.shesafe.database.EmergencyContacts;
+import com.elmeradrianv.shesafe.database.User;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import java.io.File;
+import java.util.HashMap;
+
+
+public class ProfileUserFragment extends Fragment {
+    protected EmergencyContactsCardAdapter adapter;
+    public final static int PICK_PHOTO_CODE = 1046;
+    private ParseUser currentUser = ParseUser.getCurrentUser();
+    private ParseFile profilePhoto;
+    private ImageView ivProfileView;
+    public static final String TAG = ProfileUserFragment.class.getSimpleName();
+
+    public ProfileUserFragment() {
+        // Required empty public constructor
+    }
+
+    public static ProfileUserFragment newInstance() {
+        return new ProfileUserFragment();
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_profile_user, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        setupRecycleView(view);
+        profilePhoto = ParseUser.getCurrentUser().getParseFile(User.PROFILE_PHOTO_KEY);
+        ivProfileView = view.findViewById(R.id.ivProfilePhoto);
+        putProfileImage();
+        TextView tvUsername = view.findViewById(R.id.tvUsername);
+        view.findViewById(R.id.btnChangeProfilePhoto).setOnClickListener(v -> onPickPhoto());
+        HashMap<String, MyPair<EditText, Boolean>> editUserFields = setupHashMapEditView(view);
+        fillUserFields(editUserFields);
+        HashMap<ImageButton, String> buttonsFieldsUser = setupHashMapButtons(view);
+        setupButtonsListeners(buttonsFieldsUser, editUserFields);
+        view.findViewById(R.id.btnSaveChanges).setOnClickListener(v -> setupSaveChanges(editUserFields));
+        view.findViewById(R.id.btnLogout).setOnClickListener(v -> setupLogout());
+        view.findViewById(R.id.btnAddContact).setOnClickListener(v -> setupAddContact(view));
+        view.findViewById(R.id.btnAdd).setOnClickListener(v -> setupAdd(view));
+    }
+
+    private void setupAdd(View view) {
+        String nickname=((EditText)view.findViewById(R.id.etNickname)).getText().toString();
+        String number=((EditText) view.findViewById(R.id.etPhoneNumber)).getText().toString();
+        EmergencyContacts contact = new EmergencyContacts();
```

Comment:
Let's add a TODO here to move to a "parse sublcass". Let's do the work in another PR but we can use the docs here to make this a bit easier to work with:

https://docs.parseplatform.org/android/guide/#subclasses